### PR TITLE
fix: pagination url - use regex to remove service url + leaf

### PR DESCRIPTION
**Pull request by casz**
Hmm couldn't figure out a good way to add a test :sweat:

### CODE REVIEW

File: src/infrastructure/RequestHelper.ts [hunk 1]
```diff
@@ -40,7 +40,9 @@ export async function get(
   // Rescurse through pagination results
   if (!query.page && underLimit && pagination.next) {
     const { next } = Li.parse(headers.link);
-    const more = await get(service, next.replace(`${service.url}/`, ''), {
+    const leaf = service.url.split('/').pop() || ""
```
Comment (jdalrymple) [inline src/infrastructure/RequestHelper.ts]: if the url was http://gitlab.com/api/v1, wouldnt leaf be v1?

**Comment by jdalrymple** [comment]
> Hmm couldn't figure out a good way to add a test sweat

Hmm maybe by mimicking the pagination tests with different urls ?

File: src/infrastructure/RequestHelper.ts (hunk 1 above)
Comment (casz) [reply src/infrastructure/RequestHelper.ts]: The regex handles it just fine:

See here: https://regex101.com/r/2LSP9L/1

**Comment by casz** [comment]
> Hmm maybe by mimicking the pagination tests with different urls ?

Feel free to add the tests, my mocking skills are not that good :sweat: Much better at regex :laughing:

File: src/infrastructure/RequestHelper.ts (hunk 1 above)
Comment (jdalrymple) [reply src/infrastructure/RequestHelper.ts]: In that case leaf is projects, but what if leaf were v1, like the case I pointed about above.

Comment (casz) [reply src/infrastructure/RequestHelper.ts]: My dear sir, you can modify the regex :sweat:

Comment (jdalrymple) [reply src/infrastructure/RequestHelper.ts]: Lol I realised that after I commented.

**Comment by jdalrymple** [comment]
Ill add tests later, this is gonna cause many people problems if its now pushed ASAP

**Comment by casz** [comment]
hold off :sweat:

**Comment by casz** [comment]
it is missing a single backlash for digits... because it is a string regex :cry: 

I just tested it in a playground.

**Pull request merged**